network/src/transport.rs

```diff
@@ -132,56 +143,56 @@ pub struct Connection<TSocket> {
     pub metadata: ConnectionMetadata,
 }
 
-fn identity_key_to_peer_id(
-    trusted_peers: &RwLock<HashMap<PeerId, NetworkPublicKeys>>,
-    remote_static_key: &[u8],
-) -> Option<PeerId> {
-    for (peer_id, public_keys) in trusted_peers.read().unwrap().iter() {
-        if public_keys.identity_public_key.to_bytes() == remote_static_key {
-            return Some(*peer_id);
-        }
+/// Convert a remote pubkey into a network `PeerId`.
+fn identity_pubkey_to_peer_id(
+    maybe_trusted_peers: Option<&Arc<RwLock<HashMap<PeerId, NetworkPublicKeys>>>>,
+    remote_pubkey: &x25519::PublicKey,
+) -> io::Result<PeerId> {
+    if let Some(trusted_peers) = maybe_trusted_peers {
+        // if mutual authentication, try to find peer with this identity pubkey.
+        find_peer_with_pubkey(&trusted_peers.read().unwrap(), remote_pubkey)
```

Comment:
no need to unwrap since we're returning a result

Reply:
we always unwrap locks, since we have been poisoned otherwise and need to crash the service, there is no recovering from that :/

Reply:
I had the same dilemma on the noise side, and decided that at this point you can still continue to use the current connection you have and log the fact that you're having issues creating new connections. This way we can continue running while we fix the issue.

Reply:
A poisoned lock means something else panicked while holding the lock, which means we're already in a presumably unrecoverable state (?). I'm not sure there's much value to handling this defensively while we're already crashing. If anything, instead of trying to continue in this sort of weird state, we should probably just crash the whole process and let the monitoring agent or w/e restart it...

of course, erlang has some opinions on this, about crashing actors and recovering them. there's actually an erlang inspired framework/runtime for rust https://docs.rs/bastion/0.3.4/bastion/ which might have some interesting ideas to look into.

Reply:
yeah I don't think we should assume that we're in an unrecoverable state. Maybe the debug interface panicked, and why would we care about that, etc.

I really like erlang's approach too, but it's a bit too late to transform our codebase :D we should do that for another implementation though :p